src/data/components/table/DataTable.tsx

```diff
@@ -21,215 +25,331 @@ interface DataTableColumn {
-    borderColor: '#EEEEEE',
-    fontWeight: 'inherit',
-    fontSize: 'inherit',
-    borderBottom: '1px solid #eee',
-    verticalAlign: 'top',
-  },
-  cellExpandable: {
-    cursor: 'pointer',
-    display: 'flex',
-    justifyContent: 'space-between',
-    alignItems: 'center',
-    padding: '10px',
-  },
-  expansionCell: {
-    borderColor: '#EEEEEE',
-    padding: 0,
-  },
-  bolderText: {
-    fontWeight: 'bold',
-  },
-});
-
-function Row({ rowData, columns }: DataTableRowProps) {
-  const [open, setOpen] = useState(false);
-  const classes = useRowStyles();
-
-  const DefaultCellComponent = ({ data }: DataTableCellProps) => <>{data == null ? <NoData/> : data }</>;
-
-  const cells = columns.map((column, i) => ({
-    data: rowData[i],
-    CellComponent: column.renderCell ? column.renderCell : DefaultCellComponent,
-    ExpandedComponent: column.renderExpanded ? column.renderExpanded : () => <></>,
-    ...column,
-  }));
-
-  const openStyle = open ? classes.bolderText : null;
-
-  return (
-    <>
-      <TableRow className={`${classes.root} ${classes.row}`}>
-        {cells.map((cell, i) =>
-          cell.expandable ? (
-            <TableCell
-              key={i}
-              className={[classes.cell, classes.expansionCell].join(' ')}
-            >
-              <div>
-                <div className={[classes.cellExpandable, openStyle].join(' ')} onClick={() => setOpen(!open)}>
-                  <cell.CellComponent rowData={rowData} {...cell} />
-                  {open ? <ExpandLessIcon /> : <ExpandMoreIcon />}
-                </div>
-                <Collapse in={open} timeout="auto" >
-                  {open ? <cell.ExpandedComponent rowData={rowData} {...cell} /> : null} 
-                </Collapse>
-              </div>
-            </TableCell>
+function GetCell({
+  renderCell,
+  expandable,
+  cellData,
+  renderExpanded,
+  id,
+  onChange,
+  open,
+  rowData,
+  heightChange,
+}: {
+  renderCell: createCellFunction | undefined;
+  expandable: boolean | undefined;
+  cellData: any;
+  renderExpanded: createCellFunction | undefined;
+  open: boolean;
+  onChange: (rowIndex: number, open: boolean) => void;
+  id: number;
+  rowData: any[];
+  heightChange: (rowIndex: number, height: number) => void;
+}): JSX.Element {
+  const classes = useStyles();
+  const data = cellData !== null ? cellData : '-';
+  if (expandable && renderCell && renderExpanded) {
+    return (
+      <ExtendableCell
+        renderCell={renderCell}
+        cellData={cellData}
+        renderExpanded={renderExpanded}
+        open={open}
+        onChange={onChange}
+        id={id}
+        heightChange={heightChange}
+      />
+    );
+  } else {
+    return (
+      <TableCell
+        component="div"
+        className={[classes.flexContainer, classes.noBorder].join(' ')}
+        align={'left'}
+      >
+        <div className={classes.standardSize}>
+          {renderCell !== undefined ? (
+            renderCell({ data, rowData })

```

Comment:
`renderCell` er etter hva jeg kan se en react komponent og må derfor også brukes som en react-komponent 
```suggestion
          <renderCell data={data} rowData={rowData} />
```
Den brude også ha PascalCase navn som en React  komponent